src/schedule/edf_schedule.c

```diff
@@ -64,9 +86,24 @@ static void edf_scheduler_run(void *data)
 	struct task *task;
 	uint64_t deadline;
 	uint32_t flags;
+#if CONFIG_EDF_SCHEDULER_BUDGETING
+	uint64_t systick;
+#endif
 
 	tr_dbg(&edf_tr, "edf_scheduler_run()");
 
+#if CONFIG_EDF_SCHEDULER_BUDGETING
+	timer_disable(cpu_timer_get(), edf_sch, cpu_get_id());
+
+	/* budget is given per systick, for example if systick is configured
+	 * as 1ms and budget is 8000 cycles, then task will be able to use
+	 * up to 8MCPS (or more if there are no other tasks)
+	 */
+	systick = platform_timer_get(timer_get()) /
```

Comment:
Are we basing the budget on 8000 timer_ticks ? If so, I think this would be better as a per platform CONFIG_ variable (given that non CAVS platforms will have a different timer).

Reply:
It's just an example. This is defined as CONFIG and defaults to 8k.

Reply:
ok, it's probably good to explain this in the Kconfig "help" for the option. I guess 8k is based on a % of total budget (based on max clock rate?).  Just looking for detail so that the option can be easily tuned based on teh help.

Reply:
Well it's already in help in kconfig - that it's within 1 systick IDK if we need to add more there.
It's not really % because when you switch from HPRO to LPRO you have 120k cycles in systick instead of 400k and budget for IPCs stays the same. It's just value enough to handle IPCs without stalling other tasks.